pkg/tsdb/influxdb/influxql/util/util.go

```diff
@@ -0,0 +1,154 @@
+package util
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+
+	"github.com/grafana/grafana/pkg/tsdb/influxdb/models"
+)
+
+var (
+	legendFormat = regexp.MustCompile(`\[\[([\@\/\w-]+)(\.[\@\/\w-]+)*\]\]*|\$([\@\w-]+?)*`)
+)
+
+const (
+	GraphVisType data.VisType = "graph"
+	TableVisType data.VisType = "table"
+	LogsVisType  data.VisType = "logs"
+)
+
+func FormatFrameName(rowName, column string, tags map[string]string, query models.Query, frameName []byte) []byte {
+	if query.Alias == "" {
+		return BuildFrameNameFromQuery(rowName, column, tags, frameName, query.ResultFormat)
+	}
+	nameSegment := strings.Split(rowName, ".")
+
+	result := legendFormat.ReplaceAllFunc([]byte(query.Alias), func(in []byte) []byte {
+		aliasFormat := string(in)
+		aliasFormat = strings.Replace(aliasFormat, "[[", "", 1)
+		aliasFormat = strings.Replace(aliasFormat, "]]", "", 1)
+		aliasFormat = strings.Replace(aliasFormat, "$", "", 1)
```

Comment:
If performance is a concern, I would move over the bytes and replace them in a loop. This creates a lot of strings. I haven't measured it, though :)

Reply:
I decided to create a new PR that introduces the util package. Those functions were in the code for a while already. I moved them into this package. For the performance implications, I'll have another PR.